From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 06 Mar 2016 00:40:31 +0000
Subject: thermal: Fix ABI change in 4.4.4
Forwarded: not-needed

struct thermal_zone_device is always allocated in thermal_core and the
new field is only used there.  So we can safely move it to the end of the
structure and hide it from genksyms.

Also hide the new #include, as if it could make an actual difference,
stupid genksyms...

---
--- a/include/linux/thermal.h
+++ b/include/linux/thermal.h
@@ -198,7 +198,6 @@ struct thermal_zone_device {
 	int emul_temperature;
 	int passive;
 	unsigned int forced_passive;
-	atomic_t need_update;
 	struct thermal_zone_device_ops *ops;
 	struct thermal_zone_params *tzp;
 	struct thermal_governor *governor;
@@ -208,6 +207,9 @@ struct thermal_zone_device {
 	struct mutex lock;
 	struct list_head node;
 	struct delayed_work poll_queue;
+#ifndef __GENKSYMS__
+	atomic_t need_update;
+#endif
 };
 
 /**
--- a/drivers/thermal/thermal_core.c
+++ b/drivers/thermal/thermal_core.c
@@ -37,7 +37,9 @@
 #include <linux/of.h>
 #include <net/netlink.h>
 #include <net/genetlink.h>
+#ifndef __GENKSYMS__
 #include <linux/suspend.h>
+#endif
 
 #define CREATE_TRACE_POINTS
 #include <trace/events/thermal.h>
